code/modules/ninja/martial_art.dm

```diff
@@ -0,0 +1,182 @@
+			Your grabs will instantly be aggressive while you are using this style.<br>Using middle mouse button while on harm intent and behind a person will put them in a silencing choke hold.<br>\
+			Using middle mouse button on a nearby person while on disarm intent will wrench their wrist, causing them to drop what they are holding.</span>"
+
+/datum/martial_art/ninja_martial_art/proc/wrist_wrench(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(!D.stat && !D.weakened)
+		if(has_focus)
+			has_focus = 0
+			A.face_atom(D)
+			D.visible_message("<span class='warning'>[A] grabs [D]'s wrist and wrenches it sideways!</span>", \
+							  "<span class='userdanger'>[A] grabs your wrist and violently wrenches it to the side!</span>")
+			playsound(get_turf(A), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+			D.emote("scream")
+			D.drop_item()
+			D.apply_damage(5, BRUTE, pick("l_arm", "r_arm"))
+			D.Stun(1)
+			spawn(50) has_focus = 1
+			return 1
+		A << "<span class='warning'>You are not focused enough to use that move yet!</span>"
+		return 0
+	return A.pointed(D)
+
+/datum/martial_art/ninja_martial_art/proc/choke_hold(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(!D.stat && !D.weakened)
+		A.face_atom(D)
+		if(A.dir != D.dir) // If the user's direction is not the same as the target's after A.face_atom(D) you are not behind them, and cannot use this ability.
+			A << "<span class='warning'>You cannot grab [D] from that angle!</span>"
+			return 0
+
+		if(has_choke_hold) // Are we already choking someone?
+			A<< "<span class='warning'>You are have a target in your grip!</span>"
+			return 0
+
+		has_choke_hold = 1
+
+		var/hold_name = "[pick(attack_names)] [pick("grip", "hold", "vise", "press")]"
+
+		D.visible_message("<span class='warning'>[A] comes from behind and puts [D] in a [hold_name]!</span>", \
+						  "<span class='userdanger'>[A]\ puts you in a [hold_name]! You are unable to speak!</span>")
+		step_to(D,get_step(D,D.dir),1)
+
+		D.grabbedby(A, 1)
+		var/obj/item/weapon/grab/G = A.get_active_hand()
+		if(G)
+			G.state = GRAB_NECK
+
+		var/I = 0
+		while(I < 20) // Loop to process the silence for the person being strangled so we don't have to add 20 silence all at once.
+			if(G == A.get_active_hand() && G.state >= GRAB_NECK) // Grab must be in the user's active hand for the duration of the strangle.
+				D.silent += 1
+				D.adjustOxyLoss(1)
+			else
+				D.visible_message("<span class='warning'>[A] loses his grip on [D]'s neck!</span>", \
+									"<span class='userdanger'>[A] loses his grip on your neck!</span>")
+				has_choke_hold = 0
+				return 0
+			I++
+			sleep(5)
+
+		A << "<span class='warning'>You feel [D] go limp in your grip.</span>"
+		D << "<span class='userdanger'>You feel your consciousness slip away as [A] strangles you!</span>"
+		D.sleeping += 20
```

Comment:
Sleeping isn't properly integrated into all the places it should be, exactly because the only things that use it are like.. the sleep command and the "SSDs stay asleep" stuff.